### DISCUSSION TITLE
Supporting an upgrade for redisclusters from pre 0.14.0

### DISCUSSION THREAD

**Discussion by jgatt513** [OT-CONTAINER-KIT/redis-operator#568] [Q&A]
Hello Redis team and community,

Are there plans on supporting an upgrade for redisclusters from pre 0.14.0 or is the only solution forward is to blow away all the clusters and let the operator recreate them?

**Comment by shubham-cmyk** [comment]
The v0.14.0 and v0.15.0 are incompatible due to various breaking change, so you have to uninstall and install that again and crd should also be updated ( because the scheme are not compatible )  You might try to take a backup and restore them it is supported by the operator we have written backup and restore bash scripts. 

First try it in the dev env : https://github.com/OT-CONTAINER-KIT/redis-operator/tree/master/scripts

**Reply by darkrift** [reply]
CRD incompatibility can be managed through webhook conversion

**Reply by darkrift** [reply]
This is described here : https://book.kubebuilder.io/multiversion-tutorial/tutorial.html

**Reply by shubham-cmyk** [reply]
As per this solution, we should roll out the new API version like v1beta2 for the new schema
This is a good solution. Would rolling out a v0.15.1 could be a good idea that could just support the conversion from the v0.14.0? This would solve your podSecurityContext schema problem as well
